File: backend/data/flow_graph.py
"""Allowed patient transfer routes between departments.

Based on Table 1 from the FNER game rules. A patient exiting from
department A can only be transferred to departments in FLOW_GRAPH[A].
"""

from models.enums import DepartmentId

# Allowed transfer destinations for each department
FLOW_GRAPH: dict[DepartmentId, list[DepartmentId]] = {
    DepartmentId.ER: [
        DepartmentId.SURGERY,
        DepartmentId.CRITICAL_CARE,
        DepartmentId.STEP_DOWN,
    ],
    DepartmentId.SURGERY: [
        DepartmentId.CRITICAL_CARE,
        DepartmentId.STEP_DOWN,
    ],
    DepartmentId.CRITICAL_CARE: [
        DepartmentId.SURGERY,
        DepartmentId.STEP_DOWN,
    ],
    DepartmentId.STEP_DOWN: [
        DepartmentId.SURGERY,
        DepartmentId.CRITICAL_CARE,
    ],
}


def can_transfer(from_dept: DepartmentId, to_dept: DepartmentId) -> bool:
    """Check if a transfer route is allowed."""
    return to_dept in FLOW_GRAPH.get(from_dept, [])

File: frontend/src/utils/flowGraph.ts
import type { DepartmentId } from '../types/game';

export const FLOW_GRAPH: Record<DepartmentId, DepartmentId[]> = {
  er: ['surgery', 'cc', 'sd'],
  surgery: ['cc', 'sd'],
  cc: ['surgery', 'sd'],
  sd: ['surgery', 'cc'],
};

export function canTransfer(from: DepartmentId, to: DepartmentId): boolean {
  return (FLOW_GRAPH[from] ?? []).includes(to);
}
